js/app.js

```diff
@@ -1 +1,112 @@
-console.log('Hello frontend');
+let isGameActive = true;
+
+
+document.addEventListener('DOMContentLoaded', () => {
+   
+    let squares = document.querySelectorAll('.grid-container div')
+    let playerDisplay = document.querySelector('#player')  
+   
+    
+
+    
+
+    let currentPlayer = 'Player X'
+    squares.forEach(square => {
+        square.addEventListener('click', clickOutcome)
+    });
+
+
+    function clickOutcome(e) {
+        const squareArray = Array.from(squares)
+        const index = squareArray.indexOf(e.target)
+        playerDisplay.textContent = "Your turn, " + currentPlayer
+        
+
+        if(currentPlayer === 'Player X'){
+            squares[index].classList.add('playerO')
+            currentPlayer = 'Player O'
+        } else {
+            currentPlayer = 'Player X'
+            squares[index].classList.add('playerX')
+        }
```

Comment:
You end up processing all clicks on the divs regardless of whether that square has been played before and regardless of whether someone has won already or not.  Does that make sense?  As with most of these things, there are multiple ways to solve the problem (one of them could have been implementing the clickOnce function you had below)